crates/attribute-macros/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+//! Attribute proc-macro are defined here for easy to use.
+//!
+//! It's intended to be used with the `holochain_metrics` crate like this:
+//! ```
+//! use holochain_metrics::prelude::*;
+//!
+//! #[latency]
+//! pub fn invoke_attr_macro() {
+//!     println!("Attribute Macro Invoked.");
+//!     let _x = 3 + 4;
+//! }
+
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use quote::quote;
+use syn::{parse_macro_input, parse_quote, Attribute, FnArg, Ident, ItemFn, PatType, ReturnType};
+
+#[proc_macro_attribute]
+pub fn latency(args: TokenStream, input_function: TokenStream) -> TokenStream {
```

Comment:
I kind of agree with @struktured that this could be more descriptive. I also like publish_latency.

```suggestion
pub fn publish_latency(args: TokenStream, input_function: TokenStream) -> TokenStream {
```